src/app/tests/suites/commands/interaction_model/InteractionModel.h

```diff
@@ -251,8 +251,20 @@ class InteractionModelCommands
 
             chip::app::CommandPathParams commandPath = { endpointId, clusterId, commandId,
                                                          (chip::app::CommandPathFlags::kEndpointIdValid) };
-            auto commandSender                       = std::make_unique<chip::app::CommandSender>(
-                mCallback, device->GetExchangeManager(), mTimedInteractionTimeoutMs.HasValue(), mSuppressResponse.ValueOr(false));
+            std::unique_ptr<chip::app::CommandSender> commandSender;
+            if (device->GetSecureSession().Value()->AllowsLargePayload())
+            {
+                commandSender = std::make_unique<chip::app::CommandSender>(mCallback, device->GetExchangeManager(),
+                                                                           mTimedInteractionTimeoutMs.HasValue(),
+                                                                           mSuppressResponse.ValueOr(false),
+                                                                           /* allowLargePayload = */ true);
+            }
+            else
+            {
+                commandSender = std::make_unique<chip::app::CommandSender>(mCallback, device->GetExchangeManager(),
+                                                                           mTimedInteractionTimeoutMs.HasValue(),
+                                                                           mSuppressResponse.ValueOr(false));
+            }
```

Comment:
```suggestion
            std::unique_ptr<chip::app::CommandSender> commandSender =
                std::make_unique<chip::app::CommandSender>(
                     mCallback, 
                     device->GetExchangeManager(),
                     mTimedInteractionTimeoutMs.HasValue(),
                     mSuppressResponse.ValueOr(false),
                     device->GetSecureSession().Value()->AllowsLargePayload());

```

Reply:
For what it's worth, the constructor of commandSender taking in 3 separate bool values is terrible. This should have become a params object instead so we can have defaults and make it extensible. `CommandSender(callback, exchangeManager, true,false,true)` is not readable and even if you add comments you have to ensure order and comments match.